components/kserve/kserve.go

```diff
@@ -150,9 +150,7 @@ func (k *Kserve) ReconcileComponent(cli client.Client, owner metav1.Object, dsci
 	}
 
 	if err := deploy.DeployManifestsFromPath(cli, owner, DependentPath, dscispec.ApplicationsNamespace, ComponentName, enabled); err != nil {
-		if strings.Contains(err.Error(), "spec.selector") && strings.Contains(err.Error(), "field is immutable") {
-			// ignore this error
-		} else {
+		if !strings.Contains(err.Error(), "spec.selector") || !strings.Contains(err.Error(), "field is immutable") {
```

Comment:
is this a lint error to have the original comments? 
i would keep it for a reason even the logic is flipped

Reply:
yea so lint was complaning that there was an empty block

Reply:
but then we need to put the comments in this new code.
it is difficult to maintain in long term when not knowing if this skip is done explicitly or just mistaken

Reply:
just a curious question, so previously we had the comment which said that we were ignoring the block, but in our case now, we are not ignoring the error. Instead we are going into that block and returning the error right? so what comment should we add 😅

Reply:
I can add something like `// if either of the error occurs return the error` (but that is understood from the code right :P)

Reply:
explicitly ignore error if error contains keywords "spec.selector" and "field is immutable"

Reply:
so I guess if we need to do that I can revert back to the original code and add a `//nolint` ?

Reply:
i am confused now.

isnt it just to put


		```
if !(strings.Contains(err.Error(), "spec.selector") && strings.Contains(err.Error(), "field is immutable")) {
// explicitly ignore error if error contains keywords "spec.selector" and "field is immutable" and return all other error
			return err
		}
```

Reply:
oh okay got it 👍 I was confused actually

Reply:
added the comments 👍

Reply:
is it a OR logic or AND ?

Reply:
it should be OR right?

Reply:
previous condition was `strings.Contains(err.Error(), "spec.selector") && strings.Contains(err.Error(), "field is immutable")`
so if we need to ignore only these condtions then it should be 
`!strings.Contains(err.Error(), "spec.selector") || !strings.Contains(err.Error(), "field is immutable")`
right?

Reply:
It's either `!(a && b)` or `!a || !b` if we are negating the original clause... but I think overall it's a bit confusing. What kind of error do we want to omit?

Reply:
the logic is to "explicitly ignore error if error contains keywords "spec.selector" and "field is immutable"

that's why i said just add a ! with a () is the easiest change to get lint work.

```
if !(strings.Contains(err.Error(), "spec.selector") && strings.Contains(err.Error(), "field is immutable")) {
// explicitly ignore error if error contains keywords "spec.selector" and "field is immutable" and return all other error
	return err
}

```

Reply:
but if you want to use `!() || !() `I am fine with that too.

Reply:
I would still vouch for extracting it to a function with a descriptive name - would make it easier to read the code every single time. Having it named properly you express your intent instead of asking everyone to interpret the if clause every time you stumble upon it. And we already have it in two places :)

Reply:
we can do it in a followup PR, 
i think the linter PR itself is ok to just get linter cleaned up